pages/[[...slug]]/index.tsx

```diff
@@ -0,0 +1,35 @@
+import { getAllPathnames, getPageContentByPathname } from '../../utilities/api'
+import type { GetStaticPaths, GetStaticProps } from 'next'
+import PageContent from '../../components/Main/PageContent'
+import type { PageContentProps } from '../../components/Main/PageContent'
+import React from 'react'
+import type { Pathname } from '../../utilities/api'
+
+type SitePageProps = {
+  pageContent: PageContentProps
+}
+
+export default function SitePage({ pageContent }: SitePageProps): JSX.Element {
+  return <PageContent content={pageContent.content} title={pageContent.title} />
+}
+
+export const getStaticProps: GetStaticProps = async ({ params }) => {
+  const pathname: string = (params?.slug as string[])?.join('/') ?? 'index'
```

Comment:
Annoying that we have to cast the `params?.slug` here.

Could we do something like?
```
const pathname = Array.isArray(params?.slug) ? params.slug.join('/') : 'index';
```

Reply:
I tried it and that works. However, TS threw a warning. It seems TS doesn't know we already check that `params` is defined and `slug` is an array. 
![image](https://user-images.githubusercontent.com/30286194/112219511-f7c19800-8c89-11eb-970d-4041d4676f5d.png)
So I have to add another casting to promise TS `params` must be a `ParsedUrlQuery` here. But TS keeps on complaining `slug` could be a string, which is so dumb. And then I have to promise `slug` has to be a string array. The code ended up like this:
![image](https://user-images.githubusercontent.com/30286194/112222459-6fdd8d00-8c8d-11eb-8def-821e1c24b2b0.png)

I read about the difference between type assertation and casting from [here](https://basarat.gitbook.io/typescript/type-system/type-assertion#type-assertion-vs-casting). The author deems type assertation as a way to promise TS the type has been checked. I don't know the other ways to solve these warnings and willing to know more.

Reply:
This is not a valid discussion now. Thanks for the advice.